fe/fe-core/src/main/java/com/starrocks/mysql/nio/AcceptListener.java

```diff
@@ -95,9 +95,9 @@ public void handleEvent(AcceptingChannel<StreamConnection> channel) {
                         }
                         Pair<Boolean, String> registerResult = connectScheduler.registerConnection(context);
                         if (registerResult.first) {
-                            MysqlProto.sendResponsePacket(context);
                             connection.setCloseListener(
                                     streamConnection -> connectScheduler.unregisterConnection(context));
+                            MysqlProto.sendResponsePacket(context);
```

Comment:
good catch, approved

Reply:
@nshangyiming do you elaborate this case a little bit? 

What does this condition mean?
> if (registerResult.first) {

Does it mean the we fail to register? 

And why order matters in this case?

Reply:
OK. I got it.

> if (registerResult.first) { 

it means the register connection successfully.

The reason why put `MysqlProto.sendResponsePacket(context);` before `setCloseListener` is:
- if last connection is in wrong state, calling `MysqlProto.sendResponsePacket(context);` will raise a exception
- which will lead the new connection does not `setCloseListener`.
- so when this new connection is closed, the connection number will not decrease.